src/Microsoft.Performance.SDK.Runtime/Options/Serialization/Loading/FilePluginOptionsLoader.cs

```diff
@@ -20,15 +22,37 @@ public sealed class FilePluginOptionsLoader
     /// <param name="filePath">
     ///     The path to the file from which to load options.
     /// </param>
-    public FilePluginOptionsLoader(string filePath)
-        : base(true)
+    /// <param name="logger">
+    ///     The logger to use.
+    /// </param>
+    /// <exception cref="ArgumentNullException">
+    ///     <paramref name="filePath"/> is or <paramref name="logger"/> <c>null</c>.
+    /// </exception>
+    /// <exception cref="ArgumentException">
+    ///     <paramref name="filePath"/> is empty or composed only of whitespace.
+    /// </exception>
+    public FilePluginOptionsLoader(string filePath, ILogger logger)
+        : base(true, logger)
     {
+        Guard.NotNull(filePath, nameof(filePath));
```

Comment:
>NotNull [](http://example.com/codeflow?start=14&length=7)

nit: could simplify this code a bit with Guard.NotNullOrWhiteSpace

Reply:
Forgot that existed. Will update in next PR